### ISSUE TITLE
Proposal: eslint-disable-block

### ISSUE THREAD

**Issue by StreetStrider** [eslint/eslint#5914]
Hello. I'm thinking of a feature that allows to disable a set of rules for whole JavaScript code block.

*What*: A comment ontop of block which disables rule for the rest of the block.
```js
{
  /* eslint-disable-block newline-per-chained-call */
  …
}
```

*What is the main benefit*:
* It allows to automatically enable rules at block ends, preventing you from forgetting to do it manually.
* Saves your time.
* Makes code clearer, since it will not be pollutted by closing comment.

JS block is a natural syntax for grouping elements, so I think it would be awesome to leverage it for eslint pragmas.

*Issues*: The possible issue here is how to treat rule if it lints curly braces. Should this rule work for enclosing braces or not.

I've searched through issues but found nothing even similar. Only #5206 which introduces modificator for next line.

**Comment by kaicataldo** [comment]
I'm personally not a huge fan of this idea.

> It allows to automatically enable rules at block ends, preventing you from forgetting to do it manually.
If anything, I think it makes it harder to keep track of what ESLint is or isn't ignoring, especially if the block being ignored is a large one. You'd have to check the top of every block you touch to ensure you're ESLint is linting your changes.

**Comment by StreetStrider** [comment]
@kaicataldo, that is more an argument to not to have big blocks or not to use this pragma on it. I think this feature would be good for down-level blocks which is small. In other cases off-on pragma is better.

**Comment by platinumazure** [comment]
My personal take: This proposal is redundant because the proposed scenario can also be covered with `/* eslint-disable */` and `/* eslint-enable */`. The added user benefit does not seem to be worth the extra maintenance headache (`eslint.report` is already very complicated).

But we'll see if anyone on the team wants to champion it.

**Comment by StreetStrider** [comment]
@platinumazure, OK, I'm cool with that.
The one thing I want to be noted is that the benefit is the ability to not to write closing pragma, so it works very natural, like resource deallocation when something leaves the block in languages like C++ / Rust.
